packages/site/cms-site/src/sitePreview/SitePreviewUtils.ts

```diff
@@ -67,10 +67,11 @@ export async function sitePreviewRoute(request: NextRequest, graphQLFetch: Graph
     return redirect(params.get("path") || "/");
 }
 
-export async function previewParams(): Promise<SitePreviewParams | null> {
+export async function previewParams(options: { checkDraftMode: boolean } = { checkDraftMode: true }): Promise<SitePreviewParams | null> {
     const previewScopeSigningKey = getPreviewScopeSigningKey();
 
-    if (!draftMode().isEnabled) return null;
+    // Allow skip checking for DraftMode is necessary when being called from middleware.ts (see https://github.com/vercel/next.js/issues/52080)
```

Comment:
What do you think about moving this comment to the option as JSDoc? This would result in better DX in IDEs. Also, IMO `skipDraftModeCheck` would be a better name.